core/src/components/segment-button/segment-button.tsx

```diff
@@ -79,6 +79,10 @@ export class SegmentButton implements ComponentInterface, ButtonInterface {
     };
   }
 
+  componentDidLoad() {
```

Comment:
Threading for discussion:

Are we confident this is a v7 regression? When I triaged this in https://github.com/ionic-team/ionic-framework/issues/26830#issuecomment-1437653626, I was able to reproduce this in v6, but perhaps I reproduced a different issue.

Reply:
Alternatively: Is this a bug that's always been there but is now causing issues because of v7 changes? (possibly the initializeNextTick change?)

Reply:
Ok I did a bit more digging and this is (fortunately) not an Ionic v7 regression. However, it is much more noticeable now that the `initializeNextTick` config was removed in Ionic v7.

The problem here is that if the `value` property on `ion-segment-button` is set asynchronously, the `updateState` method is not called again.

So given the following:

```html
<ion-segment value="general">
  <ion-segment-button>General</ion-segment-button>
</ion-segment>
```

If you were to asynchronously set the value of the segment button to `"general"`, the segment button would not appear checked.

Below is a test you can use to verify the fix.

<details>
<summary>segment/test/async/segment.e2e.ts</summary>

```typescript
import { expect } from '@playwright/test';
import { test } from '@utils/test/playwright';

test.describe('segment: async', () => {
  test('should set checked state when value is set asynchronously', async ({ page, skip }) => {
    skip.rtl();
    skip.mode('md');

    await page.setContent(`
      <ion-segment value="first">
        <ion-segment-button>First</ion-segment-button>
      </ion-segment>
    `);

    const segmentButton = page.locator('ion-segment-button');

    await segmentButton.evaluate((el: HTMLIonSegmentButtonElement) => el.value = 'first');
    await page.waitForChanges();

    await expect(segmentButton).toHaveClass(/segment-button-checked/);
  });
});

```
</details>

Reply:
If possible, we should merge this into `main` so v6 devs can benefit from this fix too.